nexus/db-queries/src/db/datastore/silo.rs

```diff
@@ -91,6 +92,37 @@ impl DataStore {
         new_silo_params: params::SiloCreate,
         dns_update: DnsVersionUpdateBuilder,
     ) -> CreateResult<Silo> {
+        let conn = self.pool_authorized(opctx).await?;
+        self.silo_create_conn(
+            conn,
+            opctx,
```

Comment:
I know this isn't totally new in this PR, but what's the justification for separately passing an `opctx` and a `group_opctx`?

I've always considered `OpContext` a representation of "who called this function, and with what authn/authz", so seeing two of them in a single function is a little jarring. After all, isn't only one "caller" asking for the silo to be created?

(I think this has something to do with "automatically giving admin access to a particular group within a newly created silo", I guess I'm just a little surprised to see that take the form of an `OpContext`, since I would have suspected that this group name would just be a "name" parameter. **I think passing an OpContext may be totally fine here**, but I think it's probably worth clarifying the distinction between two parameters to a function that happen to have the same type)

Reply:
Yeah, the problem here is that you have two different actors performing this operation together: (1) the end user creating the Silo, and (2) Nexus itself, acting on behalf of that same user.  The reason is that while the person is probably a Fleet Admin, they do not generally have the privilege to create a group inside the Silo (see #1580, #1681), and they certainly don't have permission to read and modify the fleet-wide DNS configuration.  But we _do_ want them to be able to do create a group and read/modify the DNS configuration _in this specific case_ that is highly controlled.  This is similar to the behavior that during provisioning, Nexus needs access to the list of Sleds in the system, but the _user_ doing the provisioning doesn't have permission to see that.  In both cases, Nexus assumes an identity _as itself_ to do those operations.  This one's a little more complicated because we need _both_ OpContexts.  We can't _just_ use Nexus's because we want to use the user's OpContext to decide whether they're allowed to do other stuff as part of this operation, like create the Silo itself.

There's a comment about this in the caller:
https://github.com/oxidecomputer/omicron/blob/2317079d26e22c78c67054e4ec7fa2a2ad5aeb79/nexus/src/app/silo.rs#L85-L88

I will update that comment to talk about DNS and add some comments here to explain this.

Reply:
Fixed in the latest commit.